src/lib/syntax/ast/token.rs

```diff
@@ -58,8 +58,8 @@ pub enum TokenData {
     Punctuator(Punctuator),
     /// A string literal
     StringLiteral(String),
-    /// A regular expression
-    RegularExpression(String),
+    /// A regular expression, consisting of body and flags
+    RegularExpression(String, String),
```

Comment:
I wonder if this should be called "RegularExpressionLiteral" to match up with the other token names?
The specification also refers to them as [RegularExpressionLiteral](https://tc39.es/ecma262/#prod-RegularExpressionLiteral)s too